recipes/libgit2/all/conanfile.py

```diff
@@ -65,6 +67,10 @@ def configure(self):
             if self.settings.os != "Windows":
                 raise ConanInvalidConfiguration("win32 is only valid on Windows")
 
+        if self.options.with_regex == "regcomp" or self.options.with_regex == "regcomp_l":
```

Comment:
Doing it this way, makes the actual value of `self.options.with_regex` available in dependencies, instead of auto)
(I think this should also be done in cmake and other recipes)
```suggestion
        self.options.with_regex = self._with_regex
        if self.options.with_regex == "regcomp" or self.options.with_regex == "regcomp_l":
```

Reply:
@remiburtin 
Could you please check this remark?

Reply:
@madebr I have already made the suggested changes. I'm going to mark the conversation as resolved.